torchmetrics/timeseries.py

```diff
@@ -0,0 +1,48 @@
+from typing import List, Tuple, Union
+
+import torch
+from torch import nn
+
+from torchmetrics.metric import Metric
+
+
+class TimeSeriesMetric(nn.ModuleList):
+    """
+    A class that keeps track of metrics as a timeseries, and implements useful methods.
+
+    Args:
+        metric: the torchmetric to keep track of at each timestep.
+    """
+
+    def __init__(self, metric: Metric):
+        super().__init__()
+        self.metric = metric
+
+    @property
+    def n_timesteps(self) -> int:
+        return len(self)
+
+    def add_timestep(self, preds: torch.Tensor, targets: torch.Tensor) -> None:
+        """Adds a metric module to the module list"""
+        module = self.metric()
+        module.update(preds, targets)
+        self.append(module)
+
+    def best_metric(self, return_timestep: bool = False)\
+            -> Union[float, Tuple[int, float]]:
+        """Returns the highest metric out of all the timesteps.
+
+        Args:
+            return_timestep: True to return the timestep with the highest metric value.
+
+        Returns:
+            The best metric value, and optionally the timestep.
+        """
+        idx, max = torch.max(self.all_metrics(), 0)
+        if return_timestep:
+            return idx.item(), max.item()
+        return max.item()
+
+    def all_metrics(self) -> torch.Tensor:
```

Comment:
since this is just calling compute of all modules I feel like it is a fitting name
```suggestion
    def compute(self) -> torch.Tensor:
```